internal/provider/webhook/datasource/datasource.go

```diff
@@ -0,0 +1,78 @@
+package datasource
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+
+	"terraform-provider-render/internal/client/webhooks"
+	"terraform-provider-render/internal/provider/webhook"
+
+	"terraform-provider-render/internal/client"
+	"terraform-provider-render/internal/provider/common"
+	rendertypes "terraform-provider-render/internal/provider/types"
+)
+
+// Ensure the implementation satisfies the expected interfaces.
+var (
+	_ datasource.DataSource              = &webhookDataSource{}
+	_ datasource.DataSourceWithConfigure = &webhookDataSource{}
+)
+
+// NewWebhookDataSource is a helper function to simplify the provider implementation.
+func NewWebhookDataSource() datasource.DataSource {
+	return &webhookDataSource{}
+}
+
+// webhookDataSource is the data source implementation.common.
+type webhookDataSource struct {
+	client  *client.ClientWithResponses
+	ownerID string
+}
+
+// Configure adds the provider configured client to the data source.
+func (d *webhookDataSource) Configure(_ context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	data := rendertypes.ConfigureDatasource(req, resp)
+	if data == nil {
+		return
+	}
+
+	d.client = data.Client
+	d.ownerID = data.OwnerID
+}
+
+// Metadata returns the data source type name.
+func (d *webhookDataSource) Metadata(_ context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_webhook"
```

Comment:
Is webhook the right name for this? Webhook is the request that is sent, but this is managing the configuration of what webhooks should be sent where.

Reply:
It is a little bit weird but we refer to webhook configuration as `webhook` in the dashboard as well and I think that's pretty standard in the industry. [GitHub does it too](https://docs.github.com/en/webhooks/using-webhooks/creating-webhooks) referring to things like `add webhook`.